.travis.yml

```diff
@@ -18,25 +18,29 @@ matrix:
   include:
    - os: linux
      language: python
+     before_install:
+      - "export DISPLAY=:0.0"
+      # set pixel depth https://docs.travis-ci.com/user/gui-and-headless-browsers/
+      - "/sbin/start-stop-daemon --start --quiet --pidfile /tmp/custom_xvfb_00.pid --make-pidfile --background --exec /usr/bin/Xvfb -- :00 -ac -screen 0 1280x1024x16"
+      - sleep 3 # give xvfb some time to start
    - os: osx
      osx_image: Xcode 8.3.3
      language: generic
+     before_install:
+      - "export DISPLAY=:0.0"
+      - Xvfb :0.0 -screen 1 1280x1024x16 &
+      - sleep 3 # give xvfb some time to start
+      - rm -rf ~/home/.enthought
+      - export PATH="/usr/local/opt/python@2/libexec/bin:$PATH"
    - os: linux
      language: python
      env: TESTENV=flake8
 
-before_install:
-  - sleep 3 #give xvfb some time to start
+before_script:
   - rm -rf ~/home/.enthought
-  - if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then export PATH="/usr/local/opt/python@2/libexec/bin:$PATH"; fi
   - make deps 
   - pip list
-
-before_script:
-  - "export DISPLAY=:0.0"
   - "export PATH=/usr/local/bin/:./:$PATH"
-  - Xvfb :0.0 -screen 1 800x600x16 &
-  - sleep 3
```

Comment:
why are we getting rid of running in the virtual screen buffer (basically our only unit test of the console)?

Reply:
With this config we get on linux:

```
Unexpected X11 display depth=8!
There's no Visual compatible with minimal AGG requirements:
At least 15-bit color depth and True- or DirectColor class.
Traceback (most recent call last):
  File "/home/travis/virtualenv/python2.7.14/lib/python2.7/site-packages/enable/qt4/base_window.py", line 198, in paintEvent
    self.handler.paintEvent(event)
  File "/home/travis/virtualenv/python2.7.14/lib/python2.7/site-packages/enable/qt4/base_window.py", line 50, in paintEvent
    self._enable_window._paint(event)
  File "/home/travis/virtualenv/python2.7.14/lib/python2.7/site-packages/enable/abstract_window.py", line 458, in _paint
    self._gc = self._create_gc(size)
  File "/home/travis/virtualenv/python2.7.14/lib/python2.7/site-packages/enable/qt4/image.py", line 23, in _create_gc
    bottom_up = 0)
  File "/home/travis/virtualenv/python2.7.14/lib/python2.7/site-packages/kiva/agg/__init__.py", line 40, in __init__
    bottom_up)
  File "/home/travis/virtualenv/python2.7.14/lib/python2.7/site-packages/kiva/agg/agg.py", line 797, in __init__
    raise ValueError(msg)
ValueError: Image depth and format are incompatible
```

https://travis-ci.org/swift-nav/piksi_tools/jobs/443883949#L1573

I have configured osx and linux specific setups in `before_install` which get rid of this error.